@here/harp-mapview/lib/geometry/TileGeometryManager.ts

```diff
@@ -22,12 +46,106 @@ export interface TileGeometryManager {
+     *
+     * @param {(GeometryKind | GeometryKind[] | GeometryKindSet)} kind The kind to add or remove
+     *      from the enabled set.
+     * @param {boolean} addOrRemoveToEnabledSet Pass in `true` to add the kind to the set, pass in
+     *      `false` to remove from that set.
+     */
+    enableKind(
+        kind: GeometryKind | GeometryKind[] | GeometryKindSet,
+        addOrRemoveToEnabledSet: boolean
+    ): void;
+
+    /**
+     * Disable a [[GeometryKind]] by adding it to the disabled set, or remove it from that set.
+     *
+     * @param {(GeometryKind | GeometryKind[] | GeometryKindSet)} kind The kind to add or remove
+     *      from the disabled set.
+     * @param {boolean} addOrRemoveToHiddenSet Pass in `true` to add the kind to the set, pass in
+     *      `false` to remove from that set.
+     */
+    disableKind(
+        kind: GeometryKind | GeometryKind[] | GeometryKindSet,
+        addOrRemoveToDisabledSet: boolean
+    ): void;
+
+    /**
+     * Hide a [[GeometryKind]] by adding it to the hidden set, or remove it from that set.
+     *
+     * @param {(GeometryKind | GeometryKind[] | GeometryKindSet)} kind The kind to add or remove
+     *      from the hidden set.
+     * @param {boolean} addOrRemoveToHiddenSet Pass in `true` to hide the kind(s), `false` to show
+     *      it again.
+     */
+    hideKind(
+        kind: GeometryKind | GeometryKind[] | GeometryKindSet,
+        addOrRemoveToHiddenSet: boolean
+    ): void;
+
+    /**
+     * Return all [[GeometryKind]]s that are contained in the tiles.
+     *
+     * @param {Tile[]} tiles The
+     * @returns {GeometryKindSet}
+     */
+    getAvailableKinds(tiles: Tile[]): GeometryKindSet;
 }
 
 /**
- * Base class for all [[TileGeometryManager]]s.
+ * Base class for all [[TileGeometryManager]]s. Handles visibility as well as enabling/disabling of
+ * kinds of geometry [[GeometryKind]].
  */
 export abstract class TileGeometryManagerBase implements TileGeometryManager {
+    get enabledGeometryKinds(): GeometryKindSet {
+        return this.enabledKinds;
+    }
+
+    set enabledGeometryKinds(kinds: GeometryKindSet) {
+        this.enabledKinds = kinds;
+    }
+
+    get disabledGeometryKinds(): GeometryKindSet {
+        return this.disabledKinds;
+    }
+
+    set disabledGeometryKinds(kinds: GeometryKindSet) {
+        this.disabledKinds = kinds;
+    }
+
+    get hiddenGeometryKinds(): GeometryKindSet {
+        return this.hiddenKinds;
+    }
+
+    set hiddenGeometryKinds(kinds: GeometryKindSet) {
+        this.hiddenKinds = kinds;
+        this.incrementVisibilityCounter();
+    }
+
+    protected get visibilityCounter(): number {
+        return this.m_visibilityCounter;
+    }
+
+    enableFilterByKind: boolean = true;
+
+    protected enabledKinds: GeometryKindSet = new GeometryKindSet();
+    protected disabledKinds: GeometryKindSet = new GeometryKindSet();
+    protected hiddenKinds: GeometryKindSet = new GeometryKindSet();
```

Comment:
Do we really need 3 sets? To simplify could we have only disabled and hidden sets and assume everything else is enabled by default? Or if it more common use case only enabled and hidden sets?
Also do we need to prevent the user from adding something to both the enable and disable sets simulatneously?

Reply:
It is not that simple, as the geometry kinds are not known beforehand, and they may vary. Basic kinds like "road" can be disabled, with only "road:bike" enabled to show only bike routes. This can be used without actually knowing the complete sets of kinds. An empty kind (enabled/disables or hidden) means no filtering.

Reply:
Ok understood, thanks for the clarification.